sycl/source/handler.cpp

```diff
@@ -319,6 +320,9 @@ event handler::finalize() {
         std::move(MAccStorage), std::move(MSharedPtrStorage),
         std::move(MRequirements), std::move(MEvents), MCGType, MCodeLoc));
     break;
+  case detail::CG::ExecCommandBuffer:
+    assert(false && "Error: Command graph submission should not be finalized");
```

Comment:
Can we throw here rather than assert? The assert won't exist in a release build

Reply:
The issue I think with throwing is that this is not an error a user should see/be able to catch, let alone be able to do anything with if they did since they would not have any control over this. Asserts are fairly commonly used throughout the runtime here for situations like this (and not including the case for this enum produces a warning, so removing it is undesirable).

I imagine that CI testing is done with asserts enabled which is where these would matter, even if they are disabled for the actual compiler releases.

Reply:
If this is consistent with the rest of the runtime, then happy to keep as-is

> I imagine that CI testing is done with asserts enabled which is where these would matter, even if they are disabled for the actual compiler releases.

Having just looked at the test-e2e suite, it relies on asserts so my assumption that asserts will not always be available could not be true.